aioesphomeapi/model.py

```diff
@@ -408,7 +408,7 @@ class LastResetType(APIIntEnum):
 @dataclass(frozen=True)
 class SensorInfo(EntityInfo):
     device_class: str = ""
-    unit_of_measurement: str = ""
```

Comment:
https://esphome.io/components/sensor/index.html
> device_class (Optional, string): The device class for the sensor. See https://developers.home-assistant.io/docs/core/entity/sensor/#available-device-classes for a list of available options. Set to "" to remove the default device class of a sensor.


It looks like the docs have the above so I'm not sure this is going to solve the problem

Reply:
In that case, for AQI the default isnt set to None, where should that happen?

Reply:
I think it will fix this specific issue but I'm worried about the side effects of this change since it could mean everyone who hasn't set a device class will now get None instead of "" 

How that is going to affect their setups is not clear here. Best case nobody notices, but it could be a significant breaking change for some.  

It's difficult to understand the consequences of this change without looking through how core and frontend handle the difference between "" and None

Reply:
For clarity, this seems like a good change, but we need to better understand the impact before moving it forward

Reply:
Agreed, I also don't know the internals, but at least it's an attempt :)

Reply:
It looks like the default was added in https://github.com/esphome/aioesphomeapi/commit/10724ea3e54f7a9cd7734b0183ba0e3e25807588

Reply:
related issue https://github.com/home-assistant/core/issues/28157

Reply:
So I don't think this will work since protobuf doesn't serialize nulls.  They need to be explicitly set as another field.

We would have to do something like
```
--- a/aioesphomeapi/api.proto
+++ b/aioesphomeapi/api.proto
@@ -259,6 +259,7 @@ message ListEntitiesBinarySensorResponse {
   bool disabled_by_default = 7;
   string icon = 8;
   EntityCategory entity_category = 9;
+  bool has_device_class = 10;
 }
 message BinarySensorStateResponse {
   option (id) = 21;

```

But if we never need `""` to be sent as a device class we can convert them to `None` instead explicitly in the integration

Reply:
>  the device class for AQI should be None instead of an empty string.

I just realized  this is actually about `unit_of_measurement`

Reply:
I think https://github.com/home-assistant/core/pull/96556 will solve this

Let me know what you think.